src/libraries/Common/src/System/Net/Http/aspnetcore/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -92,7 +93,7 @@ internal async ValueTask SetSecurityConfigForConnection(X509Certificate cert, st
             MsQuicParameterHelpers.SetSecurityConfig(MsQuicApi.Api, _ptr, (uint)QUIC_PARAM_LEVEL.CONNECTION, (uint)QUIC_PARAM_CONN.SEC_CONFIG, _securityConfig!.NativeObjPtr);
         }
 
-        internal override IPEndPoint RemoteEndPoint => new IPEndPoint(_remoteEndPoint.Address, _remoteEndPoint.Port);
+        internal override EndPoint RemoteEndPoint => _remoteEndPoint;
```

Comment:
Shouldn't we create a copy of the endpoint here?

Reply:
We should if this was a public API, but this is internal now so I didn't bother.